src/pynwb/image.py

```diff
@@ -73,19 +79,104 @@ def __init__(self, **kwargs):
         if unit is None:
             kwargs['unit'] = ImageSeries.DEFAULT_UNIT
 
+        if (
+            args_to_set["external_file"] is not None
+            and args_to_set["starting_frame"] is None
+        ):
+            args_to_set["starting_frame"] = (
+                [0] if len(args_to_set["external_file"]) == 1 else None
+            )
+
         super().__init__(**kwargs)
 
-        if args_to_set["external_file"] is None:
-            args_to_set["starting_frame"] = None  # overwrite starting_frame
         for key, val in args_to_set.items():
             setattr(self, key, val)
 
+        if not self._check_external_file_starting_frame_length():
+            msg = ("%s '%s': The number of frame indices in 'starting_frame' should have "
+                   "the same length as 'external_file'." % (self.__class__.__name__, name))
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
+        if self._change_external_file_format():
+            warnings.warn(
+                "%s '%s': The value for 'format' has been changed to 'external'. "
+                "Setting a default value for 'format' is deprecated and will be changed "
+                "to raising a ValueError in the next release."
+                % (self.__class__.__name__, self.name),
+                DeprecationWarning,
+            )
+
+        if not self._check_external_file_format():
+            msg = (
+                "%s '%s': Format must be 'external' when external_file is specified."
+                % (self.__class__.__name__, name)
+            )
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
+        if not self._check_external_file_data():
+            msg = (
+                "%s '%s': Either external_file or data must be specified (not None), but not both."
+                % (self.__class__.__name__, name)
+            )
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
         if not self._check_image_series_dimension():
             warnings.warn(
                 "%s '%s': Length of data does not match length of timestamps. Your data may be transposed. "
-                "Time should be on the 0th dimension" % (self.__class__.__name__, self.name)
+                "Time should be on the 0th dimension"
+                % (self.__class__.__name__, self.name)
             )
 
+    def _raise_error_when_check_is_violated(self, error_msg):
```

Comment:
Instead of ``_raise_error_when_check_is_violated`` I would suggest renaming the function to ``_error_on_new_warn_on_construct`` just to be more explicit on the purpose of the function. T